utils/testsuite/testsuite.py

```diff
@@ -195,7 +197,7 @@ def __str__(self):
 
 
 TIMEOUT_COMPILER = 200
-TIMEOUT_VM = 200
+TIMEOUT_VM = 600
```

Comment:
This change seems odd, are you seeing tests that need 10 minutes to run? If so, we should skip them and add a TODO to investigate why they are timing out.

Reply:
The top 10 slowest tests from the most recent passed run of test-linux-test262 job are:

```sh
Top 10 slowest tests
test262/test/built-ins/decodeURI/S15.1.3.1_A2.5_T1.js                     388.806
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A2.5_T1.js            387.991
test262/test/built-ins/RegExp/character-class-escape-non-whitespace.js    117.804
test262/test/built-ins/decodeURI/S15.1.3.1_A1.12_T2.js                     52.211
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.12_T2.js            51.000
test262/test/built-ins/decodeURI/S15.1.3.1_A1.12_T1.js                     49.002
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.12_T3.js            48.802
test262/test/built-ins/decodeURI/S15.1.3.1_A1.12_T3.js                     48.693
test262/test/built-ins/decodeURI/S15.1.3.1_A1.2_T1.js                      48.508
test262/test/built-ins/decodeURI/S15.1.3.1_A1.11_T1.js                     48.404
```

The Linux VM being used to run the job seems to be quite limited in CPU and memory. The longest test ends up taking 388+ seconds.

Reply:
If this is purely an issue for the linux job, let's just configure the build with `-O2` so the test runs faster.

Reply:
Updated PR to add `-DCMAKE_CXX_FLAGS=-O2  -DCMAKE_C_FLAGS=-O2` to cmake configure command for test-linux-test262 CircleCI job. I have not yet reduced the timeout value. Let's see how long the slowest tests take now on CircleCI, then I'll reduce the timeout value.

Reply:
Thanks for the guidance @neildhar. With the compiler optimization flag, the top 10 slowest tests are now:

```
Top 10 slowest tests
test262/test/built-ins/decodeURI/S15.1.3.1_A2.5_T1.js                                      30.295
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A2.5_T1.js                             29.000
test262/test/intl402/String/prototype/localeCompare/returns-same-results-as-Collator.js     7.703
test262/test/built-ins/RegExp/character-class-escape-non-whitespace.js                      6.098
test262/test/intl402/Date/prototype/returns-same-results-as-DateTimeFormat.js               4.307
test262/test/built-ins/decodeURI/S15.1.3.1_A1.11_T1.js                                      3.498
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.12_T1.js                             3.407
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.11_T2.js                             3.299
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.12_T3.js                             3.298
test262/test/built-ins/decodeURIComponent/S15.1.3.2_A1.11_T1.js                             3.290
```

I have updated the PR to revert the time out value change.